internal/grpc/services/usershareprovider/usershareprovider.go

```diff
@@ -497,40 +506,181 @@ func (s *service) GetReceivedShare(ctx context.Context, req *collaboration.GetRe
 }
 
 func (s *service) UpdateReceivedShare(ctx context.Context, req *collaboration.UpdateReceivedShareRequest) (*collaboration.UpdateReceivedShareResponse, error) {
-
-	if req.Share == nil {
+	if req.GetShare() == nil {
 		return &collaboration.UpdateReceivedShareResponse{
 			Status: status.NewInvalid(ctx, "updating requires a received share object"),
 		}, nil
 	}
-	if req.Share.Share == nil {
+
+	if req.GetShare().GetShare() == nil {
 		return &collaboration.UpdateReceivedShareResponse{
 			Status: status.NewInvalid(ctx, "share missing"),
 		}, nil
 	}
-	if req.Share.Share.Id == nil {
+
+	if req.GetShare().GetShare().GetId() == nil {
 		return &collaboration.UpdateReceivedShareResponse{
 			Status: status.NewInvalid(ctx, "share id missing"),
 		}, nil
 	}
-	if req.Share.Share.Id.OpaqueId == "" {
+
+	if req.GetShare().GetShare().GetId().GetOpaqueId() == "" {
 		return &collaboration.UpdateReceivedShareResponse{
 			Status: status.NewInvalid(ctx, "share id empty"),
 		}, nil
 	}
 
+	isStateTransitionShareAccepted := slices.Contains(req.GetUpdateMask().GetPaths(), _fieldMaskPathState) && req.GetShare().GetState() == collaboration.ShareState_SHARE_STATE_ACCEPTED
+	if isStateTransitionShareAccepted {
+		gatewayClient, err := s.gatewaySelector.Next()
+		if err != nil {
+			return nil, err
+		}
+
+		receivedShare, err := gatewayClient.GetReceivedShare(ctx, &collaboration.GetReceivedShareRequest{
+			Ref: &collaboration.ShareReference{
+				Spec: &collaboration.ShareReference_Id{
+					Id: req.GetShare().GetShare().GetId(),
+				},
+			},
+		})
+		switch {
+		case err != nil:
+			fallthrough
+		case receivedShare.GetStatus().GetCode() != rpc.Code_CODE_OK:
+			return &collaboration.UpdateReceivedShareResponse{
+				Status: receivedShare.GetStatus(),
+			}, err
+		}
+
+		resourceStat, err := gatewayClient.Stat(ctx, &provider.StatRequest{
+			Ref: &provider.Reference{
+				ResourceId: receivedShare.GetShare().GetShare().GetResourceId(),
+			},
+		})
+		switch {
+		case err != nil:
+			fallthrough
+		case resourceStat.GetStatus().GetCode() != rpc.Code_CODE_OK:
+			return &collaboration.UpdateReceivedShareResponse{
+				Status: resourceStat.GetStatus(),
+			}, err
+		}
+
```

Comment:
You could use `utils.GetResource` instead of all these lines.

Reply:
utils.GetResourceByID returns an error instead of the status...

```go
...
if err := checkStatusCode("getting resource", res.GetStatus().GetCode()); err != nil {
  return nil, err
}
...
```

i dont like that approach, since in my opinion the rpc endpoint should only if something network related or similar is off... what do you think?

Reply:
I prefer simplicity. We don't need to check explicitly for a network error here. Both is just an (unrecoverable) error for us. So I would opt for using `utils.GetResource`. 

But not critical of course if you prefer the other way.

Reply:
i would like to keep it as it is... since a status that is off is not an error in my opinion